semgrep-core/src/engine/Match_patterns.ml

```diff
@@ -109,7 +109,7 @@ let match_sts_sts rule a b env =
             | [] -> env
             | stmt :: _ -> MG.extend_stmts_match_span stmt env
           in
-          GG.m_stmts_deep ~less_is_ok:true a b env))
+          GG.m_stmts_deep ~inside:rule.R.inside ~less_is_ok:true a b env))
```

Comment:
Can we put this in the env?

Reply:
Oh I see `env` is something different. Leaving the comment because I still think this feels a little spaghetti.

Reply:
Why spaghetti, due to the number of arguments being passed? Eventually we could perhaps pass a record.